programs/Makefile

```diff
@@ -235,8 +235,16 @@ zstdmt: zstd
 	ln -sf zstd zstdmt
 
 .PHONY: generate_res
-generate_res:
-	windres/generate_res.bat
+generate_res: $(RES64_FILE) $(RES32_FILE)
+
+ifneq (,$(filter Windows%,$(OS)))
+RC ?= windres
+# http://stackoverflow.com/questions/708238/how-do-i-add-an-icon-to-a-mingw-gcc-compiled-executable
+$(RES64_FILE): windres/zstd.rc
+	$(RC) -o $@ -I ../lib -I windres -i $< -O coff -F pe-x86-64
+$(RES64_FILE): windres/zstd.rc
```

Comment:
is it really RES64_FILE ? not RES32_FILE ?

Reply:
Good catch

Reply:
you can also use a variable that store pe-x86-64 or pe-i386 (where you set RES_FILE) to avoid a check here

Reply:
`make generate_res` builds both for some reason, so I'll keep it as is (with fix).